node/src/test/scala/coop/rchain/finalization/FinalizationSpec.scala

```diff
@@ -71,61 +71,65 @@ class FinalizationSpec extends FlatSpec with Matchers {
     }
 
     def runRandom = {
-
       val nets = List(10).map(genNet)
+      nets.traverse { case (net, name) => randomTest(net, name) }
+    }
 
-      nets.traverse {
-        case (net, name) =>
-          for {
-            net1_     <- runSections(net, List((1, .0f)), s"start-$name")
-            (net1, _) = net1_
+    def runInfinite: F[Unit] = {
+      val nets = List(10).map(genNet)
+      nets.tailRecM[F, Unit] { networks =>
+        val newNetState = networks.traverse {
+          case (net, name) => randomTest(net, name).map { case (newNet, _) => (newNet, name) }
+        }
+        newNetState.map(_.asLeft[Unit]) // Infinite loop
```

Comment:
We need some restriction because tests on CI must finish. Later we can create infinite versions outside of tests that runs as part of CI.

Reply:
I [added](https://github.com/rchain/rchain/pull/3601/commits/888b61149c65dd93ccb1da90a95e8814cd36683d) ignoring the infinite test by default. Also, I added the ability to turn off output for the infinite test, and now it works faster without generating images and polluting the output.